gm4_vecto_shamir/data/gm4_vecto_shamir/functions/vehicle/check.mcfunction

```diff
@@ -0,0 +1,5 @@
+#run from vehicle/spawn_(boat/minecart) and vehicle/check
+
+execute as @e[tag=gm4_ml_vehicle] at @a[tag=gm4_ml_invehicle,predicate=gm4_vecto_shamir:not_in_vehicle] if score @s gm4_ml_vecto_id = @p[tag=gm4_ml_invehicle,predicate=gm4_vecto_shamir:not_in_vehicle,distance=..1,sort=nearest] gm4_ml_vecto_id run function gm4_vecto_shamir:vehicle/remove
+execute unless entity @e[tag=gm4_ml_vehicle] run tag @a remove gm4_ml_invehicle
```

Comment:
I think the best way to do this is to tag the player when they summon a boat/minecart and then have a clock check if they have the tag, but are not in a temp vehicle, then from there kill all boats/minecarts that have the same `gm4_ml_vecto_id` score as the player.

Reply:
> this removes the tag `gm4_ml_invehicle` tag from all players? Is this intentional?

That was indeed intentional, because no player is supposed to have the tag if no temp vehicles exist.

the `gm4_ml_invehicle` tag was meant as "this player is supposed to be in a temp vehicle. If not, kill the temp vehicle with their id"

Reply:
> I think the best way to do this is to tag the player when they summon a boat/minecart and then have a clock check if they have the tag, but are not in a temp vehicle, then from there kill all boats/minecarts that have the same `gm4_ml_vecto_id` score as the player.

From what I can see, that's exactly what this function does here. In the next commit I'll rename the tag so it's a bit more clear what it does.

When a boat/minecart is summoned, the player gets a tag and the check loop starts, which keeps running until no temp vehicles exist